lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -798,7 +798,8 @@ void FMemModuleOp::insertPorts(ArrayRef<std::pair<unsigned, PortInfo>> ports) {
 
 static void buildModule(OpBuilder &builder, OperationState &result,
                         StringAttr name, ArrayRef<PortInfo> ports,
-                        ArrayAttr annotations, bool withAnnotations = true) {
+                        ArrayAttr annotations, ArrayAttr layers,
+                        bool withAnnotations = true, bool withLayers = false) {
```

Comment:
Seems like withLayers should be true by default, and every FModuleLike op except for classes+extclasses should have a layers attribute. Classes, which can never have layer dependencies, shouldn't have an attribute, and should just have a getter that returns the empty array attribute. See `ClassOp::getPortAnnotationsAttr` for example.

As well, I think getLayers/getLayersAttr should be added to the FModuleLike op interface.